draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -2712,28 +2710,121 @@ of the counter byte I2OSP(i, 1)).
 
 The essential difference between the construction of {{CDMP05}} and
 expand\_message\_xmd is that the latter hashes a counter appended to
-strxor(b\_0, b\_(i - 1)) (step 9) rather than to b\_0.
+strxor(b\_0, b\_(i - 1)) (step 10) rather than to b\_0.
 This approach increases the Hamming distance between inputs to different
 invocations of H, which reduces the likelihood that nonidealities in H
 affect the distribution of the b\_i values.
 
+## Domain separation recommendations {#security-considerations-domain-separation}
+
+The expand\_message variants in this document ({{hashtofield-expand}})
+always append a suffix-free-encoded domain separation tag DST\_prime
+to the strings hashed by H, the underlying hash or extensible output function.
+For protocols that also invoke H outside of hash\_to\_field, those invocations
+MUST be domain separated from all uses of H inside hash\_to\_field, and SHOULD
+be domain separated from uses of H outside of the protocol.
+
+The RECOMMENDED method of ensuring domain separation is appending a tag
+distinct from DST\_prime to all inputs to H outside of hash\_to\_field.
+This ensures that distinct invocations of H have distinct suffixes.
+If further separation is required among invocations of H outside of
+hash\_to\_field, protocols should use multiple tags that are
+all distinct from one another and from DST\_prime.
+
+For protocols that use expand\_message\_xmd ({{hashtofield-expand-xmd}}),
+either of the following two methods MAY be used instead.
+The first method MAY also be used with expand\_message\_xof
+({{hashtofield-expand-xof}}), but the second method MUST NOT be
+used with expand\_message\_xof.
+
+1. For each invocation of H outside hash\_to\_field, choose a unique domain
+   separation tag DST\_ext.
+   Prepend DST\_ext to the input to H, and in addition append a single
+   zero byte to this input.
+
+   Appending the byte I2OSP(0, 1) to all inputs to H outside hash\_to\_field
+   ensures that these inputs are distinct from those inside hash\_to\_field
+   because the final byte of DST\_prime encodes the length of DST, which is
+   required to be nonzero ({{domain-separation}}), and DST\_prime is always
+   appended to invocations of H inside hash\_to\_field.
+   Further, distinct DST\_ext values ensure that invocations of H outside
+   hash\_to\_field are distinct from one another.
+
+   For example, for two invocations of H whose inputs are msg1 and msg2,
```

Comment:
```suggestion
   For example, for two uses of H whose inputs are msg1 and msg2,
```

Reply:
To me, an invocation is a single call to H, e.g., H(msg), whereas (as I understand it) what we mean here is that we want to clone H twice using DST_ext1 and DST_ext2.